api/v1alpha1/runner_types.go

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MIT
+
+package v1alpha1
+
+import (
+	commonParams "github.com/cloudbase/garm-provider-common/params"
+	"github.com/cloudbase/garm/params"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// RunnerSpec defines the desired state of Runner
+type RunnerSpec struct {
+	ID string `json:"id,omitempty"`
+}
+
+// RunnerStatus defines the observed state of Runner
+type RunnerStatus struct {
+	// ID is the database ID of this instance.
+	ID string `json:"id,omitempty"`
+
+	// PeoviderID is the unique ID the provider associated
+	// with the compute instance. We use this to identify the
+	// instance in the provider.
+	ProviderID string `json:"providerId,omitempty"`
+
+	// AgentID is the github runner agent ID.
+	AgentID int64 `json:"agentId"`
+
+	// Name is the name associated with an instance. Depending on
+	// the provider, this may or may not be useful in the context of
+	// the provider, but we can use it internally to identify the
+	// instance.
+	Name string `json:"name,omitempty"`
+
+	// OSType is the operating system type. For now, only Linux and
+	// Windows are supported.
+	OSType commonParams.OSType `json:"osType,omitempty"`
+
+	// OSName is the name of the OS. Eg: ubuntu, centos, etc.
+	OSName string `json:"osName,omitempty"`
+
+	// OSVersion is the version of the operating system.
+	OSVersion string `json:"osVersion,omitempty"`
+
+	// OSArch is the operating system architecture.
+	OSArch commonParams.OSArch `json:"osArch,omitempty"`
+
+	// Addresses is a list of IP addresses the provider reports
+	// for this instance.
+	Addresses []commonParams.Address `json:"addresses,omitempty"`
+
+	// Status is the status of the instance inside the provider (eg: running, stopped, etc)
+	Status commonParams.InstanceStatus `json:"status,omitempty"`
+
+	// RunnerStatus is the github runner status as it appears on GitHub.
+	InstanceStatus params.RunnerStatus `json:"instanceStatus,omitempty"`
+
+	// PoolID is the ID of the garm pool to which a runner belongs.
+	PoolID string `json:"poolId,omitempty"`
```

Comment:
can we also get the pool Name (name of the corresponding CR) - as handling the pool IDs from a user perspective might be super ugly :sweat_smile: